dio/lib/src/options.dart

```diff
@@ -109,48 +112,48 @@ class BaseOptions extends _RequestConfig {
       connectTimeout: connectTimeout ?? this.connectTimeout,
       receiveTimeout: receiveTimeout ?? this.receiveTimeout,
       sendTimeout: sendTimeout ?? this.sendTimeout,
-      extra: extra ?? Map.from(this.extra ?? {}),
-      headers: headers ?? Map.from(this.headers ?? {}),
-      responseType: responseType ?? this.responseType,
+      extra: extra ?? this.extra,
+      headers: headers ?? this.headers,
+      responseType: (responseType != null) ? responseType : this.responseType,
```

Comment:
You can revert the change on this line, using `??` is just syntactic sugar for what you wrote.